### DISCUSSION TITLE
Intel integrated graphics clarification needed

### DISCUSSION THREAD

**Discussion by erwinkramer** [henrygd/beszel#358] [Support] [answered]
For https://beszel.dev/guide/gpu it states 
> The Docker agent does not support GPU monitoring. You must use the binary agent. 

and
> Intel GPUs are not currently supported for two reasons:

But does this count for integrated graphics as well? For example, the integrated Intel® UHD Graphics 600 is what i want to monitor. As Intel also produces dedicated GPUs now this might lead to confusion.

**Comment by henrygd** [comment, marked as answer]
It's the same for integrated graphics.

If you have an AMD iGPU, it will show usage / power of the iGPU.

With Intel, as of now you will only get CPU usage with no additional info about the iGPU.

I will update the docs to be clearer when I have time.

**Comment by erwinkramer** [comment]
Thanks! Do you perhaps know of a container that does?

**Reply by henrygd** [reply]
Looks like Netdata supports frequency and power through `intel_gpu_top`. No sure if it works in a container.

https://learn.netdata.cloud/docs/collecting-metrics/hardware-devices-and-sensors/intel-gpu

There doesn't seem to be an Intel equivalent of `nvidia-smi` / `rocm-smi` that provides the same information, unfortunately.

**Comment by oregonpillow** [comment]
@henrygd why can't iGPU stats be added? intel_gpu_top provides quite a few useful metrics including transcoding, 3d rendering, iGPU package power and more. It can also be queried from a container because i already use an intel iGPU exporter (will post below)


```sh
sudo intel_gpu_top -J

{
        "period": {
                "duration": 37.162895,
                "unit": "ms"
        },
        "frequency": {
                "requested": 0.000000,
                "actual": 0.000000,
                "unit": "MHz"
        },
        "interrupts": {
                "count": 26.908560,
                "unit": "irq/s"
        },
        "rc6": {
                "value": 99.997842,
                "unit": "%"
        },
        "power": {
                "GPU": 0.000000,
                "Package": 23.116332,
                "unit": "W"
        },
        "engines": {
                "Render/3D/0": {
                        "busy": 0.000000,
                        "sema": 0.000000,
                        "wait": 0.000000,
                        "unit": "%"
                },
                "Blitter/0": {
                        "busy": 0.000000,
                        "sema": 0.000000,
                        "wait": 0.000000,
                        "unit": "%"
                },
                "Video/0": {
                        "busy": 0.000000,
                        "sema": 0.000000,
                        "wait": 0.000000,
                        "unit": "%"
                },
                "Video/1": {
                        "busy": 0.000000,
                        "sema": 0.000000,
                        "wait": 0.000000,
                        "unit": "%"
                },
                "VideoEnhance/0": {
                        "busy": 0.000000,
                        "sema": 0.000000,
                        "wait": 0.000000,
                        "unit": "%"
                }
        },
        "clients": {

        }
```

**Reply by oregonpillow** [reply]
for example:

```yml
intel-gpu-exporter:
    devices:
      - "/dev/dri:/dev/dri"
    privileged: true
    ports:
      - "9101:9100"
    restart: unless-stopped
    container_name: intel-gpu-exporter
    image: andrewgolikov55/intel-gpu-exporter
```


![image](https://github.com/user-attachments/assets/4810fae8-3445-4967-9430-f1716973bdb2)

**Reply by henrygd** [reply]
We're still trying to figure out the best way to do it. The latest discussion is in this PR: https://github.com/henrygd/beszel/pull/755

Right now the best option seems to be `intel_gpu_top` with a line chart including the engines, but if it's possible to use oneAPI level zero with [purego](https://github.com/ebitengine/purego) then that would be better. I don't have have an Intel GPU so I'm a bit hamstrung here.

Looks like `intel-gpu-exporter` uses `intel_gpu_top` and only exports the video and blitter engines? Is this something you customized, or does that mean the Render/3D engine is unnecessary? There's also some differences I've noticed in `intel_gpu_top` output between devices that I'm not entirely clear on. But I'll look at what `intel-gpu-exporter` is doing when I have time.
